vllm/model_executor/models/mixtral.py

```diff
@@ -141,8 +142,9 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
                                         selected_experts,
                                         inplace=True)
 
-        final_hidden_states = tensor_model_parallel_all_reduce(
-            final_hidden_states)
+        if self.tp_size > 1:
```

Comment:
I'm curious what the thoughts on this are. On the one hand there is a ton of value in being able to run all the layers in a single process without having to stand up the distributed environment, on the other hand, this is redundant with the check in `tensor_model_parallel_all_reduce` that checks for `get_tensor_model_parallel_world_size() == 1` (but we can't use that here since `get_tensor_model_parallel_world_size` already needs the distributed environment. I considered monkey patching that function but that doesn't seem great either.